Sort hostnames (help needed)

skttl opened the pull request:
#6591

This is what I got so far.

I compare the order of hostnames between the submitted list, and the persisted list, and deletes the existing domains, if the order doesn't match.

As mentioned in https://github.com/umbraco/Umbraco-CMS/issues/6591#issuecomment-539188057 there is some ordering of the domains overriding the database order. I tried removing that, but that didn't help on the ordering in the links group on the info tab. I'll keep digging.

I'm not sure how to wrap the deletion/re-adding of domains in a transaction, it just uses the domainservice for updating, like before. Any pointers is welcome :)

nul800sebastiaan commented:
Aha, I am sure the Service will take care of all that transaction stuff for you, but we'll give it a look as soon as this is out of draft! 👍

skttl commented:
I added the UI, and removed the ordering from various places. But they still don't follow the db order when going through `UrlProviderExtensions.GetContentUrls` (https://github.com/umbraco/Umbraco-CMS/blob/v8/dev/src/Umbraco.Web/Routing/UrlProviderExtensions.cs#L11)

I stepped through the method, and the yields come in the right order, but somehow the end result is out of order. I need some help from this point...

nul800sebastiaan commented:
Hi @skttl - sorry for the delay on this one, it's been on the todo list for a while.

Unfortunately, we won't be able to consider a PR for this feature if it's not possible to do it without a breaking change. The feedback I got from the team is that you won't be able to reliably get the order from the DB. 

So unfortunately, I will have to close the PR and will put the issue on the ideas list for future consideration when we are allowed to make breaking changes.
